backend/server/src/models/question.rs

```diff
@@ -0,0 +1,131 @@
+use crate::models::error::ChaosError;
+use anyhow::{bail, Result};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use sqlx::{Pool, Postgres, QueryBuilder, Row};
+
+#[derive(Deserialize, Serialize)]
+pub struct Question {
+    id: i64,
+    title: String,
+    description: Option<String>,
+    required: bool,
+
+    #[serde(flatten)]
+    question_type: QuestionType,
+
+    campaign_id: i64,
+    created_at: DateTime<Utc>,
+    updated_at: DateTime<Utc>,
+}
+
+/// An enum that represents all the data types of question data that CHAOS can handle.
+/// This stores all the data for each question type.
+///
+/// \
+/// Some question types are stored in memory and JSON using the same struct, and only differ
+/// in their implementation when inserting to the database and in their restrictions
+/// (e.g. max 1 answer allowed in multi-choice vs. many in multi-select)
+///
+/// \
+/// With the chosen serde representation and the use of `#[serde(flatten)]`, the JSON for a
+/// `Question` will look like this:
+/// ```json
+/// {
+///   "title": "What is your favourite language?",
+///   "required": true,
+///   "question_type": "MultiChoice",
+///   "data": [
+///     "Rust",
+///     "Java",
+///     "TypeScript"
+///   ]
+/// }
+/// ```
+#[derive(Deserialize, Serialize)]
+#[serde(tag = "question_type", content = "data")]
+pub enum QuestionType {
+    ShortAnswer,
+    MultiChoice(MultiOptionData),
+    MultiSelect(MultiOptionData),
+    DropDown(MultiOptionData),
+}
+
+pub trait QuestionData {
+    async fn insert_into_db(self, question_id: i64, pool: &Pool<Postgres>) -> Result<i64>;
+    async fn get_from_db(self, question_id: i64, pool: &Pool<Postgres>) -> Result<QuestionType>;
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct MultiOptionData {
+    options: Vec<String>,
```

Comment:
Each option is not just a `String`, it should also have an `id`, so updating the text will update on all applications.